configuration/config-txt/boot.md

```diff
@@ -99,12 +99,37 @@ If `disable_splash` is set to `1`, the rainbow splash screen will not be shown o
 
 On the Raspberry Pi 4B, if this value is set to `0` then the interrupts will be routed to the ARM cores using the legacy interrupt controller, rather than via the GIC-400. The default value is `1`.
 
+## os_prefix
+<a name="os_prefix">
+
+`os_prefix` is an optional setting that allows you to choose between multiple versions of the kernel and Device Tree files installed on the same card. Any value in `os_prefix` is prepended to (stuck in front of) the name of any operating system files loaded by the firmware, where "operating system files" is defined to mean kernels, initramfs, cmdline.txt, .dtbs and overlays. The prefix would commonly be a directory name, but it could also be part of the filename such as "test-". For this reason, directory prefixes must include the trailing `/` character.
+
+In an attempt to reduce the chance of a non-bootable system, the firmware first tests the supplied prefix value for viability - unless the expected kernel and .dtb can be found at the new location/name, the prefix is ignored (set to ""). A special case of this viability test is applied to overlays, which will only be loaded from `${os_prefix}${overlay_prefix}` (where the default value of [`overlay_prefix`](#overlay_prefix) is "overlays/") if `${os_prefix}${overlay_prefix}README` exists, otherwise it ignores `os_prefix` and treats overlays as shared.
```

Comment:
I wonder if it's worth somehow cross-linking this to https://github.com/raspberrypi/documentation/blob/master/configuration/config-txt/boot.md where the "expected kernel names" are described?

On a side-note, should https://github.com/raspberrypi/documentation/blob/master/configuration/boot_folder.md say something about kernel filenames?? @JamesH65

Reply:
> On a side-note, should https://github.com/raspberrypi/documentation/blob/master/configuration/boot_folder.md say something about kernel filenames?? @JamesH65

Yes, I'll do that today.

They are AIUI:
kernel.img       -> Up to BCM2836? (Pi0, PI1, Pi2 with 2836)
kernel7.img     -> 2837 (Pi2 with 2837,  Pi3)
kernel7l.img    -> 2711 (Pi4), standard 32bit kernel with LPAE 
kernel8.img     -> Beta 64bit kernel

Reply:
Almost... https://github.com/raspberrypi/documentation/blob/master/configuration/config-txt/boot.md#kernel

Reply:
Bit confused about kernel.img and its use. Not sure that linked page is correct (IIRC, I wrote it...)

Reply:
I'll leave cross-linking to the experts.

Reply:
@JamesH65 Do you think it's worth making the "expected kernel and .dtb" text in the bit above a link to https://github.com/raspberrypi/documentation/blob/master/configuration/boot_folder.md :grey_question: